### PR TITLE
Update interface to reduce CSP following delay

### PR DESCRIPTION
**Pull request by nardavin**
This PR updates the interface drivers to use both position and velocity setpoints instead of just position setpoints. This reduces the following delay of the controller to approximately 0.03 seconds. This PR also reorganizes the code structure, moving many of the implementation-specific details from miso_robot/synchronized_motion to this repo. This therefore relies on [PR 15 in miso_robot](https://github.com/MisoRobotics/miso_robot/pull/15).

### CODE REVIEW

File: festo_ethercat/include/festo_ethercat/festo_cmmt_interface.h [hunk 1]
```diff
@@ -240,6 +240,7 @@ class FestoCMMTInterface
   int speed_override_percent_;         /**< Modifier for the velocity (%) */
   CiA402StateMachine state_machine_;   /**< Mirrors/controls the CiA402 state machine on the Festo CMMT. */
   uint32_t velocity_actual_mm_s_;      /**< The last velocity reading from the rail (mm/s). */
+  uint32_t profile_velocity_requested_mm_s_; /**< The maximum profile velocity at which this library is trying to move the rail (mm/s). */
```
Comment (zzv2) [inline festo_ethercat/include/festo_ethercat/festo_cmmt_interface.h:243]: might want to adjust the spacing here to be consistent

File: festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    ROS_ERROR("rail_name not set, exiting...");
+    return false;
+  }
+  if(!robot_hw_nh.getParam("rail_width_mm", rail_width_mm))
+  {
+    ROS_ERROR("rail_width_mm not set, exiting...");
+    return false;
+  }
+
+  if(rail_driver_.initializeCMMT(ethercat_interface.c_str(), (uint32_t)default_velocity, rail_width_mm))
```
Comment (zzv2) [inline festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp:40]: this returns false when it starts correctly?

File: festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  {
+    ROS_ERROR("Failed to set up ethercat slaves.");
+    return false;
+  }
+
+  while(!rail_driver_.startCyclicPositioning())
+  {
+    ros::Duration(0.01).sleep();
```
Comment (zzv2) [inline festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp:48]: better convention is to use `rate.sleep()`

File: festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp (hunk 2 above)
Comment (nardavin) [reply festo_ethercat/src/festo_ethercat/festo_hw_interface.cpp:40]: The function has a c-style integer return, where 0 is successful any anything nonzero is an error code. I'll update the if statement to make this more obvious

**Review by zzv2 — APPROVED** [review]

**Pull request merged**